src/java/io/bazel/rulesscala/worker/GenericWorker.java

```diff
@@ -1,117 +0,0 @@
-package io.bazel.rulesscala.worker;
-
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-import com.google.devtools.build.lib.worker.WorkerProtocol.WorkRequest;
-import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.PrintStream;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.List;
-
-public class GenericWorker {
-  protected final Processor processor;
-
-  public GenericWorker(Processor p) {
-    processor = p;
-  }
-
-  protected void setupOutput(PrintStream ps) {
-    System.setOut(ps);
-    System.setErr(ps);
-  }
-
-  // Mostly lifted from bazel
-  private void runPersistentWorker() throws IOException {
-    PrintStream originalStdOut = System.out;
-    PrintStream originalStdErr = System.err;
-
-    while (true) {
-      try {
-        WorkRequest request = WorkRequest.parseDelimitedFrom(System.in);
-        if (request == null) {
-          break;
-        }
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        int exitCode = 0;
-
-        try (PrintStream ps = new PrintStream(baos)) {
-          setupOutput(ps);
-
-          try {
-            processor.processRequest(request.getArgumentsList());
-          } catch (Exception e) {
-            e.printStackTrace();
-            exitCode = 1;
-          }
-        } finally {
-          System.setOut(originalStdOut);
-          System.setErr(originalStdErr);
-        }
-
-        WorkResponse.newBuilder()
-            .setOutput(baos.toString())
-            .setExitCode(exitCode)
-            .build()
-            .writeDelimitedTo(System.out);
-        System.out.flush();
-      } finally {
-        System.gc();
```

Comment:
Why drop this? I'm guessing bazel people added this after they learned it the hard way

Reply:
I'll add this call back.

Reply:
Added it back!